swarm/netsim/http.go

```diff
@@ -17,45 +17,34 @@
 package netsim
 
 import (
-	"context"
 	"fmt"
 	"net/http"
 
 	"github.com/ethereum/go-ethereum/log"
 	"github.com/ethereum/go-ethereum/p2p/simulations"
 )
 
-func (s *Simulation) initHTTPServer(opts *SimulationOptions) {
-	//assign default port if nothing provided
-	if opts.HTTPSimPort == "" {
-		opts.HTTPSimPort = DefaultHTTPSimPort
+// Package defaults.
+var (
+	DefaultHTTPSimAddr = ":8888"
+)
+
+func (s *Simulation) startHTTPServer(addr string) {
+	//assign default addr if nothing provided
+	if addr == "" {
+		addr = DefaultHTTPSimAddr
 	}
-	log.Info(fmt.Sprintf("Initializing simulation server on 0.0.0.0:%s...", opts.HTTPSimPort))
+	log.Info(fmt.Sprintf("Initializing simulation server on 0.0.0.0%s...", addr))
```

Comment:
`0.0.0.0` should be removed. If addrs is `127.0.0.1:8888`, for example, it will log invalid address.